### PR TITLE
[143] - Block negative slippage

### LINKED ISSUE #143: Handle negative slippage
The app breaks when handling negative slippage. I believe it happens also in the original Uniswap.

We should handle the error, but also probably allow negative slippage to allow to create market orders in a hacky way.

1. Click in the gear icon, for advanced settings
2. Input a negative slippage

### PR DESCRIPTION
**Pull request by W3stside**
Closes #143 

Handles/disables entering negative slippage

### CODE REVIEW

**Review by anxolin — COMMENTED** [review]
Can you point out what was the fix? I'm lost. almost 300 lines, which part of `src/custom/components/TransactionSettings/index.tsx` is ours?

**Comment by W3stside** [comment]
> Can you point out what was the fix? I'm lost. almost 300 lines, which part of `src/custom/components/TransactionSettings/index.tsx` is ours?

you gotta find it :mag:

**Comment by W3stside** [comment]
> Can you point out what was the fix? I'm lost. almost 300 lines, which part of `src/custom/components/TransactionSettings/index.tsx` is ours?

https://github.com/gnosis/gp-swap-ui/blob/0b8743d24b431d06c8528c95dc37013c48d1f6e5/src/custom/components/TransactionSettings/index.tsx#L53

**Review by alfetopito — APPROVED** [review]
Awesome, it works.
For benchmark, I tried https://app.uniswap.org/#/swap and can reproduce the error there :D

Not related to this issue, but the UI is a bit off
screenshot_2021-02-18_11-08-44

Not for this PR, just raising the point

File: src/custom/components/TransactionSettings/index.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import React, { useState, useRef, useContext } from 'react'
```
Comment (anxolin) [inline src/custom/components/TransactionSettings/index.tsx:1]: This is too big of a copy paste for such a small change

**Review by anxolin — CHANGES_REQUESTED** [review]
I think we shouldn't duplicate all this code for just this line
Ideally, it should be easy to know what we have changed

Also, the Mod version doesn't compare line by line. You deleted lot's of things:
image


Please, review the rules we follow for overriding in our document.

**Comment by anxolin** [comment]
Please, david consider this https://github.com/gnosis/gp-swap-ui/pull/166

**Comment by W3stside** [comment]
closing in favour of #166

**Pull request closed without merging**